File: dongwook/week4/덧칠하기.java
package Study_codingtest.dongwook.week4;

class 덧칠하기 {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int idx = 0;

        while (idx < section.length) {
            int start = section[idx];
            int end = start + m - 1;
          
            while (idx < section.length && section[idx] <= end) {
                idx++;
            }

            answer++;
        }

        return answer;
    }
}

File: seongmin/week4/덧칠하기/덧칠하기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, int m, vector<int> section) {
    int count = 0;
    int lastPainted = 0;

    for (const auto& pos : section) {
        if (pos > lastPainted) {
            lastPainted = pos + m - 1;
            count++;
        }
    }

    return count;
}
